crates/turbopack-ecmascript/src/path_visitor.rs

```diff
@@ -29,15 +29,23 @@ fn find_range<'a, 'b>(
     index: usize,
 ) -> Option<&'b [(&'a AstPath, &'a dyn VisitorFactory)]> {
     // Precondition: visitors is never empty
-    let start = if visitors.first().unwrap().0[index] >= *kind {
-        // Fast path: It's likely that the whole range is selected
-        0
-    } else {
-        visitors.partition_point(|(path, _)| path[index] < *kind)
-    };
+    if visitors.first().unwrap().0[index] > *kind {
+        // Fast path: If ast path of the first visitor is already out of range, then we
+        // can skip the whole visit.
+        return None;
+    }
+
```

Comment:
There could be another fast path for `visitors.last().unwrap().0[index] < *kind`

Reply:
I would still keep the old fast path too (after the out of bounds checks you added):

```
    if visitors.first().unwrap().0[index] >= *kind {
        // Fast path: It's likely that the whole range is selected
        0
    }
```

It avoids the `O(log n)` partition_point for that case.

Reply:
Looks great! Done @sokra